packages/api/src/utils/decoder.ts

```diff
@@ -16,16 +16,17 @@ export function decodeContractError(contract: ethers.Contract, errorData: string
       .join(', ');
   }
 
-  throw new Error(`${errorFragment.name} | ${message ? message : ''}`);
+  return new Error(`${errorFragment.name} | ${message ? message : ''}`);
 }
 
 export function handleError(contract: ethers.Contract, error: any) {
   if (error.error?.data?.data) {
```

Comment:
This `if` chain is merely to locate the `data` to use, yet each branch has, essentially the same statement within. 
Wouldn't null coalescing be better? Or an `if` chain that picks 1 data and then invokes the required function once.